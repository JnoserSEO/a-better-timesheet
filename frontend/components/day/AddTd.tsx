import React from 'react'
import { isToday } from 'date-fns'
import { ReducerAction } from '../timesheet'
import { DayEntry } from '../../models/fullweek.model'
import '../../styles/day/add.scss'

interface AddTdProps { 
    index: number; 
    day: DayEntry; 
    dispatch: React.Dispatch<ReducerAction>
}

const AddTd: React.FC<AddTdProps> = ({ index,day,dispatch }) => {
    console.log(day)
    return <td key={index} className={`add-td${isToday(day.date) ? ' add-today-highlight' : ''}`}>
        <div className="add-div">
            <span className='add' onClick={() => dispatch({ type: 'SHOW_MODAL', value: day })}>+</span>
        </div>
    </td>
}


export default AddTd