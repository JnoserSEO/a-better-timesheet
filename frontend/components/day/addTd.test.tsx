import React from 'react'
import { shallow } from 'enzyme'
// import toJSON from 'enzyme-to-json'
import AddTd from './AddTd'


test('should render Day correctly with no entries', () => {
    const wrapper = shallow(<AddTd index={0} day={{dayOfWeek:'Tuesday',date:new Date(1601401562455),data:[]}} dispatch={(x)=>x} />)
    expect(wrapper).toMatchSnapshot()
})