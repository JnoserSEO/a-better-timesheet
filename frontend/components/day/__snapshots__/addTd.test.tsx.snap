// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Day correctly with no entries 1`] = `ShallowWrapper {}`;
